### DISCUSSION TITLE
How can I remove the File Edit Temrminal Help menu and its submenus

### DISCUSSION THREAD

**Discussion by pingbaohua** [eclipse-theia/theia#14055] [Q&A]
How can I remove the File Edit Temrminal Help menu and its submenus

**Comment by msujew** [comment]
Hey @pingbaohua,

you can use the `MenuRegistry#unregisterMenu` API for that. For the exact menu path that needs to be supplied, see:

https://github.com/eclipse-theia/theia/blob/004e2b2c7f0b881b97f52dc84441f2abfb569590/packages/terminal/src/browser/terminal-frontend-contribution.ts#L61-L62

**Reply by pingbaohua** [reply]
there is no unregisterMenu API, i cannot find it

**Reply by msujew** [reply]
See 

https://github.com/eclipse-theia/theia/blob/4c7ffb632b46f52841f09e6fdc88e9c23448b572/packages/core/src/common/menu/menu-model-registry.ts#L197-L249

**Reply by pingbaohua** [reply]
![image](https://github.com/user-attachments/assets/b9adb4b1-6a3c-40bb-8c6b-1467e05e7e56)

What if it's these? I tried to comment bind but failed，
![image](https://github.com/user-attachments/assets/27d99a25-e452-4d71-b53a-499732325969)

**Reply by msujew** [reply]
Take a look at the [contribution filter API](https://theia-ide.org/docs/contribution_filter/) to remove those.

**Reply by pingbaohua** [reply]
I try to hide Search extention, I try to increase SearchFilterAndCommandContribution, but it is invalid，How do I achieve this

![image](https://github.com/user-attachments/assets/c07d7df8-3525-4cd5-8dad-3570d1a9e0bf)
![image](https://github.com/user-attachments/assets/a74d541b-497b-45d5-8381-013112e310b8)
